test/unit/org/apache/cassandra/index/sai/disk/v1/kdtree/KDTreeIndexBuilder.java

```diff
@@ -292,7 +293,7 @@ protected Pair<ByteComparable, IntArrayList> computeNext()
                 assertTrue(terms.hasNext());
 
                 final ByteSource encoded = TypeUtil.asComparableBytes(terms.next(), type, TypeUtil.BYTE_COMPARABLE_VERSION);
-                return Pair.create(v -> encoded, postings);
+                return Pair.create(ByteComparable.preencoded(TypeUtil.BYTE_COMPARABLE_VERSION, ByteSourceInverse.readBytes(encoded)), postings);
```

Comment:
Suggestion for future: maybe we could further simplify this. Looks like we are creating far too many temporaries.

Reply:
This is left for a follow-up as above.